dev_guide/builds.adoc

```diff
@@ -191,6 +191,77 @@ See the xref:../creating_images/s2i.adoc#creating-images-s2i[S2I Requirements] t
 on how to create a builder image supporting incremental builds.
 ====
 
+
+[[extended_builds]]
+
+=== Extended builds
+
+For compiled languages (Go, C, C++, Java, etc.) the dependencies necessary for
+compilation might dramatically increase the size of the image or introduce
+vulnerabilities that can be exploited.
+
+To avoid these problems, source-to-image introduces a dual image build process
+that allows an application to be built as usual - in the builder image, and
+then inject - in the runtime image, the resulted binary artifacts into the
+desired location to be executed.
+
+In order to offer even more flexibility in this process, S2I will execute an
+`assemble-runtime` script inside the runtime image that will allow you to
+customize further the runtime image.
+
+Official documentation for source-to-image extended builds can be found
+https://github.com/openshift/source-to-image/blob/master/docs/runtime_image.md[here].
+
+====
+[source,yaml]
+----
+strategy:
+  type: "Source"
+  sourceStrategy:
+    from:
+      kind: "ImageStreamTag"
+      name: "resulted-image:latest"
+    runtimeImage: <1>
+      kind: "ImageStreamTag"
+      name: "runtime-image:latest"
+    runtimeArtifacts: <2>
+      sourcePath: "/path/to/source"
+      destinationDir: "path/to/destination"
+----
+<1> The runtime image that the artifacts should be copied to.
+This will be the final image that the application will run on. This image
+should contain the minimum application dependencies to run the injected builder
+image binary.
+
+<2> The runtime artifacts is a mapping of resulted artifacts in the builder
+image that should be injected inside the runtime image.
+`sourcePath` can be the full path to a binary or directory inside the builder
+image.
+`destinationDir` must be a directory inside the runtime image where the
+artifacts should be copied. This directory is relative to the specified
+`WORKDIR` inside that image.
+
+====
+
+[NOTE]
+====
+The `incremental` option is not valid with `runtimeImage` because you cannot
+have incremental extended builds.
+====
+
+=== Testing your application
+
+Extended builds offers two ways of running tests against your application.
+
+The first option is to install all test dependencies and run the tests inside
+your `builder image` since that image, in the context of extended builds, will
+not be pushed to a registry.
+
+The second option is to specify a script via the post commit hook where you can
+download and install all the test dependencies. Since the tests run based on
+the runtime image but inside an ephemeral container, the downloaded
+dependencies do not make it to the runtime image.
+
```

Comment:
@php-coder btw, in the source-to-image docs there is no information about runtimeAuthentication either.